src/lib/structures/game/StandardGame.ts

```diff
@@ -0,0 +1,152 @@
+import type { Message, Snowflake, User } from 'discord.js';
+import { Game, AcceptedAnswer, GameData, GameType } from '#game/Game';
+import { BrandingColors } from '#utils/constants';
+import { createEmbed } from '#utils/responses';
+import { container } from '@sapphire/framework';
+
+export class StandardGame extends Game {
+	public gameType: GameType.Standard = GameType.Standard;
+
+	public constructor(data: GameData) {
+		super(data);
+	}
+
+	public async guess(message: Message) {
+		const guess = message.content.toLowerCase();
+		const guessedBefore = this.guessedThisRound();
+		const guessedNow = this.processGuess(guess, message.author.id);
+		const promises: Promise<unknown>[] = [];
+
+		if (!guessedNow) {
+			return;
+		}
+
+		const isHalfGuessedNow = this.acceptedAnswer === AcceptedAnswer.Both && !guessedBefore && guessedNow;
+
+		const halfGuessedString = isHalfGuessedNow
+			? ` **"${message.content}"** is the ${guessedNow.toLowerCase()}'s name. You're halfway there!`
+			: '';
+
+		const embed = createEmbed(`✅ You got it!${halfGuessedString}`);
+
+		promises.push(message.channel.send({ embeds: [embed] }));
+
+		if (!isHalfGuessedNow) {
+			promises.push(this.queue.player.stop());
+		}
+
+		await Promise.all(promises);
+	}
+
+	public async onTrackEnd() {
+		const playersGuessedSong = this.round.guessedSong;
+		const playersGuessedArtist = this.round.guessedArtists.get(this.round.primaryArtist)!;
+		// Ensure players are only in here once
+		const uniqueGuessers = [...new Set([...playersGuessedSong, ...playersGuessedArtist])];
+		const guessers = (await Promise.all(
+			uniqueGuessers.map((id) => container.client.users.fetch(id).catch(() => null)).filter(Boolean)
+		)) as User[];
```

Comment:
This is to make it more readable by separating it into 2 lines, but also I made a mistake, you should run the filter *after* all the promises have been resolved.
```suggestion
		const promises = uniqueGuessers.map((id) => container.client.users.fetch(id).catch(() => null));
		const guessers = (await Promise.all(promises)).filter(Boolean) as User[];
```